rskj-core/src/test/java/co/rsk/RskContextTest.java

```diff
@@ -43,25 +60,150 @@ public void getCliArgsSmokeTest() {
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        Files.createDirectory(testDatabasesDirectory.resolve("unitrie"));
+
+        TrieStore trieStore = rskContext.getTrieStore();
+        Assert.assertThat(trieStore, is(instanceOf(MultiTrieStore.class)));
+        Assert.assertThat(Files.list(testDatabasesDirectory).count(), is(numberOfEpochs));
+        Assert.assertThat(Files.list(testDatabasesDirectory).noneMatch(p -> p.getFileName().toString().equals("unitrie")), is(true));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    @Ignore("Permissions set fails under CircleCI")
+    public void shouldFailIfCannotBuildMultiTrieStore() throws IOException {
+        Path testDatabasesDirectory = Files.createTempDirectory("test", PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("---------")));
+        doReturn(new GarbageCollectorConfig(true, 1000, 3)).when(testProperties).garbageCollectorConfig();
+        doReturn(testDatabasesDirectory.toString()).when(testProperties).databaseDir();
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        rskContext.getTrieStore();
     }
 
-    private void mockBuildInfoResource(InputStream buildInfoStream) {
-        mockStatic(RskContext.class);
-        ClassLoader classLoader = mock(ClassLoader.class);
-        when(classLoader.getResourceAsStream("build-info.properties")).thenReturn(buildInfoStream);
-        when(RskContext.class.getClassLoader()).thenReturn(classLoader);
+    @Test
+    public void mergeMultiTrieStoreDBs() throws IOException {
+        Path testDatabasesDirectory = databaseDir.getRoot().toPath();
+
+        List<Path> sourcePaths = new ArrayList<>();
+        Set<byte[]> sourceKeys = new HashSet<>();
+        int sourcesCount = 3;
+        for (int i = 0; i < sourcesCount; i++) {
+            Path sourcePath = testDatabasesDirectory.resolve(String.format("src-%d", i));
+            sourcePaths.add(sourcePath);
+            KeyValueDataSource originDataSource = RskContext.makeDataSource(sourcePath);
+            byte[] currentElement = {(byte) i};
+            sourceKeys.add(currentElement);
+            originDataSource.put(currentElement, currentElement);
+            originDataSource.close();
+        }
+
+        Path destination = testDatabasesDirectory.resolve("destination");
+        RskContext.mergeMultiTrieStoreDBs(destination, sourcePaths);
+        KeyValueDataSource destinationDataSource = RskContext.makeDataSource(destination);
+        try {
+            Set<byte[]> destinationKeys = destinationDataSource.keys();
```

Comment:
I would try to build two maps and probe that both contains the same elements. In the first part of the test you are only checking one side of inclusion all destination items in sum(sources). You also need to probe that items in sum(sources) are included in destination.

Reply:
The test is built inserting a single element into each of the _origin_ datasources, I added an extra assertion for the keys after merge to be exactly the same count as _origin_ datasources. Then I check for the value to be the expected one (the same as key, this was made just to simplify the test). I think the whole PR is already pretty covered, but if you think it's really important to improve this particular test, could you suggest a more explicit code change (a diff would be optimal)? I may have not understood your request really well.